### PR TITLE
Lock/v1

### PR DESCRIPTION
**Pull request by ThaBrad**
- Lock V1
- ExpertMode
- Swap Screen return-path Bugfix

/kind

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] Your UI implementation visually matched the rendered design*

### CODE REVIEW

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/LOCK/LockDashboardScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+
+export interface TransactionCache {
+  amount: number
+  depositAddress: string
+  token: WalletToken
+  network: NetworkName
+  transaction?: OceanTransaction
+}
+
+export function LockDashboardScreen ({ route }: Props): JSX.Element {
```
Comment (davidleomay) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/LOCK/LockDashboardScreen.tsx:55]: What about the ToDos in this file?

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/LOCK/LockKycScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        {/* <SubmitButtonGroup
+          isDisabled={!formState.isValid}
+          label={translate('LOCK/LockDashboardScreen', 'CONTINUE')}
+          // processingLabel={translate('components/Button', 'CONTINUE')}
+          onSubmit={onSubmit}
+          title='sell_continue'
+          isProcessing={isSubmitting}
+          displayCancelBtn={false}
+          lock
+        /> */}
```
Comment (davidleomay) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/LOCK/LockKycScreen.tsx:114]: Will this button be used for new KYC?

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/LOCK/LockContextProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  const [isKycComplete, setIsKycComplete] = useState(false)
+  const [internalStakingInfo, setInternalStakingInfo] = useState<StakingOutputDto>()
+
+  /**
+   * @param lockUserDto {LockUserDto} gets LOCK's KYC state
+   * @return boolean
+   */
+
+  const setKycComplete = useCallback((lockUserDto?: LockUserDto): void => {
+    setIsKycComplete(lockUserDto === undefined || lockUserDto?.kycStatus === 'Full')
```
Comment (davidleomay) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/LOCK/LockContextProvider.tsx:27]: KYC status `Light` is enough

File: shared/api/dfx/AuthService.ts [hunk 4]
```diff
@@ -51,30 +56,50 @@ export class Session implements ISession {
 }
 
 class AuthServiceClass {
+  // TODO: (thabrad) check if able to remove!
   private readonly session$ = new ReplaySubject<Session>()
+  private dfxApiHook: (DFXAPIContextI) | undefined
```
Comment (davidleomay) [inline shared/api/dfx/AuthService.ts:61]: Is this the React workaround, you told me about?

File: shared/contexts/DFXAPIContextProvider.tsx [hunk 5]
```diff
@@ -81,6 +87,32 @@ export function DFXAPIContextProvider (props: PropsWithChildren<{}>): JSX.Elemen
...
+    //     console.log('\n\n', filteredPairList)
+
+    //     const x = pairs.filter((p) => p[0].includes('DFXWALLET.address_list'))
+    //     console.log('------------')
+    //     console.log('address_list ', JSON.parse(x.pop()?.[1] ?? ''))
+    //     console.log('------------')
+    //   })
+    // })
+    // // AuthService.updateSession({ accessToken: undefined })
+  }
```
Comment (davidleomay) [inline shared/contexts/DFXAPIContextProvider.tsx:114]: What is this for?

File: shared/contexts/DFXAPIContextProvider.tsx [hunk 6]
```diff
@@ -151,70 +183,79 @@ export function DFXAPIContextProvider (props: PropsWithChildren<{}>): JSX.Elemen
...
+        // console.log('sig: ', sig)
+        return sig
+      }
       await DFXPersistence.setPair({
         network: networkName,
         addr: address,
         signature: sig
       })
       await DFXPersistence.resetPin()
+      return sig
```
Comment (davidleomay) [inline shared/contexts/DFXAPIContextProvider.tsx:227]: Does the pin reset still work correctly?

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/LOCK/LockContextProvider.tsx (hunk 3 above)
Comment (davidleomay) [reply mobile-app/app/screens/AppNavigator/screens/Portfolio/LOCK/LockContextProvider.tsx:27]: Already resolved

File: mobile-app/app/contexts/FeatureFlagContext.tsx [hunk 7]
```diff
@@ -37,14 +38,26 @@ export function FeatureFlagProvider (props: React.PropsWithChildren<any>): JSX.E
...
-      const modFlag: FeatureFlag = { ...flag, stage: 'beta' }
-      return modFlag
-    } else {
-      return flag
-    }
-  })
+
+  // removes duplicate flags by only leaving in last (non-'public') flag
+  const deleteDuplicates = (featureFlags: FeatureFlag[]): FeatureFlag[] => {
+    const ids: FEATURE_FLAG_ID[] = ['loan', 'auction', 'dfi_loan_payment', 'local_storage', 'dusd_vault_share', 'dusd_loan_payment', 'future_swap', 'service_provider', 'onboarding_v2', 'dusd_dex_high_fee', 'dusd_dfi_high_fee', 'setting_v2', 'lock']
```
[the remark this answers is not in the document]
Comment (ThaBrad) [reply mobile-app/app/contexts/FeatureFlagContext.tsx:44]: this represents all the current available ”native" feature flags, which is used to filter for multiple stages for the same feature (expert > beta > release)

File: shared/api/dfx/AuthService.ts (hunk 4 above)
Comment (ThaBrad) [reply shared/api/dfx/AuthService.ts:61]: yup!

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/LOCK/LockContextProvider.tsx (hunk 3 above)
Comment (ThaBrad) [reply mobile-app/app/screens/AppNavigator/screens/Portfolio/LOCK/LockContextProvider.tsx]: thx

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/LOCK/LockKycScreen.tsx (hunk 2 above)
Comment (ThaBrad) [reply mobile-app/app/screens/AppNavigator/screens/Portfolio/LOCK/LockKycScreen.tsx:114]: nah, can be removed

Comment (ThaBrad) [reply mobile-app/app/screens/AppNavigator/screens/Portfolio/LOCK/LockKycScreen.tsx:114]: new kyc method is already in the screen, just disabled

File: shared/contexts/DFXAPIContextProvider.tsx (hunk 6 above)
Comment (ThaBrad) [reply shared/contexts/DFXAPIContextProvider.tsx:227]: why shouldn't it?

File: mobile-app/app/contexts/FeatureFlagContext.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (davidleomay) [reply mobile-app/app/contexts/FeatureFlagContext.tsx:44]: Ok. Would also be possible by simply grouping 😉

File: shared/contexts/DFXAPIContextProvider.tsx (hunk 5 above)
Comment (davidleomay) [reply shared/contexts/DFXAPIContextProvider.tsx:114]: Cool

**Pull request merged**